packages/dart/noports_core/lib/src/srv/srv_impl.dart

```diff
@@ -118,6 +123,148 @@ class SrvImplExec implements Srv<Process> {
   }
 }
 
+@visibleForTesting
+class SrvImplInline implements Srv<SSHSocket> {
+  final AtSignLogger logger = AtSignLogger('SrvImplInline');
+
+  @override
+  final String host;
+
+  @override
+  final int streamingPort;
+
+  @override
+  final int localPort = -1;
+
+  @override
+  final bool bindLocalPort = false;
+
+  @override
+  final String? rvdAuthString;
+
+  @override
+  final String? sessionAESKeyString;
+
+  @override
+  final String? sessionIVString;
+
+  SrvImplInline(
+    this.host,
+    this.streamingPort, {
+    this.rvdAuthString,
+    this.sessionAESKeyString,
+    this.sessionIVString,
+  }) {
+    if ((sessionAESKeyString == null && sessionIVString != null) ||
+        (sessionAESKeyString != null && sessionIVString == null)) {
+      throw ArgumentError('Both AES key and IV are required, or neither');
+    }
+  }
+
+  @override
+  Future<SSHSocket> run() async {
+    DataTransformer? encrypter;
+    DataTransformer? decrypter;
+
+    if (sessionAESKeyString != null && sessionIVString != null) {
+      final DartAesCtr algorithm = DartAesCtr.with256bits(
+        macAlgorithm: Hmac.sha256(),
+      );
+      final SecretKey sessionAESKey =
+          SecretKey(base64Decode(sessionAESKeyString!));
+      final List<int> sessionIV = base64Decode(sessionIVString!);
+
+      encrypter = (Stream<List<int>> stream) {
+        return algorithm.encryptStream(
+          stream,
+          secretKey: sessionAESKey,
+          nonce: sessionIV,
+          onMac: (mac) {},
+        );
+      };
+      decrypter = (Stream<List<int>> stream) {
+        return algorithm.decryptStream(
+          stream,
+          secretKey: sessionAESKey,
+          nonce: sessionIV,
+          mac: Mac.empty,
+        );
+      };
+    }
+
+    try {
+      logger.info('Creating socket connection to rvd at $host:$streamingPort');
+      Socket socket = await Socket.connect(host, streamingPort);
+
+      // Authenticate if we have an rvdAuthString
+      if (rvdAuthString != null) {
+        logger.info('authenticating');
+        socket.writeln(rvdAuthString);
+        await socket.flush();
+      }
+
+      WrappedSSHSocket sshSocket =
+          WrappedSSHSocket(socket, rvdAuthString, encrypter, decrypter);
+
+      return sshSocket;
+    } catch (e) {
+      AtSignLogger('srv').severe(e.toString());
+      rethrow;
+    }
+  }
+}
+
+/// - Get a hold of the underlying SSHSocket's Stream and StreamSink
+/// - Wrap the StreamSink with encrypter
+/// - Wrap the Stream with decrypter
+class WrappedSSHSocket implements SSHSocket {
```

Comment:
SSHClient requires an SSHSocket, so we create one like this, which also takes care of ensuring the outbound / inbound streams are encrypted / decrypted as needed